src/services/services.ts

```diff
@@ -2478,6 +2482,51 @@ export function createLanguageService(
         }
     }
 
+    function getLinkedEditingAtPosition(fileName: string, position: number): LinkedEditingInfo | undefined {
+        const sourceFile = syntaxTreeCache.getCurrentSourceFile(fileName);
+        const token = findPrecedingToken(position, sourceFile);
+        if (!token) return undefined;
+
+        if (isJsxFragment(token.parent.parent)) {
+            const openPos = token.parent.parent.openingFragment.getStart(sourceFile) + 1; // "<".length
+            const closePos = token.parent.parent.closingFragment.getStart(sourceFile) + 2; // "</".length
+
+            // only allows mirroring right after opening bracket: <| ></| >
+            if ((position !== openPos) && (position !== closePos)) return undefined;
+
+            const ranges = [{ start: openPos, length: 0 }, { start: closePos, length: 0 }];
+            const wordPattern = undefined;
+            return { ranges, wordPattern };
+        }
+        else {
+            // looks for mirroring in element tag names
+            const tag = findAncestor(token,

```

Comment:
If you feed in `token.parent` instead of `token` itself, the entire body can get rid of all the `parent.parent` references and talk only about `n`. Then you don't have to create a separate `element` variable below.